atomicapp/nulecule/container.py

```diff
@@ -0,0 +1,54 @@
+import os
+import shutil
+import subprocess
+import uuid
+import logging
+
+from atomicapp.constants import APP_ENT_PATH
+
+logger = logging.getLogger(__name__)
+
+
+class DockerHandler(object):
+    """Interface to interact with Docker."""
+
+    def __init__(self, docker_cli='/usr/bin/docker'):
+        self.docker_cli = docker_cli
+
+    def pull(self, image, update=False):
+        if not self.is_image_present(image) or update:
+            logger.debug('Pulling Docker image: %s' % image)
+            pull_cmd = [self.docker_cli, 'pull', image]
```

Comment:
@vpavlin I agree that we need to be able to distinguish between the formats (appc, oci, etc.. could all be options). The *we will need to change that at some point in the future* part was to indicate that we need to stop hardcoding docker and start supporting these other image formats.